src/pages/modules/components/DeleteDialog.tsx

```diff
@@ -0,0 +1,142 @@
+  marginTop: theme.spacing(0),
+  marginBottom: theme.spacing(0),
+  paddingLeft: theme.spacing(3),
+}));
+
+// weird workaround since header variants had different shades of black
+const Text = styled(Typography)(({ theme }) => ({
+  color: theme.palette.common.black,
+}));
+
+const Warning = styled(Typography)(({ theme }) => ({
+  marginTop: theme.spacing(1),
+  color: theme.palette.error.main,
+}));
+
+const Footer = styled("div")({
+  float: "right",
+});
+
+interface DeleteDialogProps {
+  open: boolean;
+  handleClose: () => void;
+  toDelete: IAdminModule;
+}
+
+export function DeleteDialog(props: DeleteDialogProps) {
+  const { open, handleClose, toDelete } = props;
+
+  const dispatch = useAppDispatch();
+
+  const handleDialogSubmit = () => {
+    dispatch(requestDeleteModule(toDelete._id));
+    // TODO:
+    //  dont close before checking
+    //  if action was successful
+    handleClose();
+  };
+
+  const handleClickAway = (event: any) => {
+    // necessary to stop propagation
+    // that cause the accordion to open/close
+    if (event) event.stopPropagation();
+    handleClose();
+  };
+
+  return (
+    <Dialog
+      onClose={(event) => handleClickAway(event)}
+      open={open}
+      maxWidth="md"
+    >
+      <Paper elevation={3}>
+        <DialogTitle>Confirm Module Deletion</DialogTitle>
+        <Divider />
+
+        <DialogContent>
+          <Text variant="h5" gutterBottom>
+            Attempting to delete: "Module {toDelete.number}: {toDelete.name}"
+          </Text>
+
+          {toDelete.problems.length > 0 ? (
+            <>
+              <Text variant="h6">
+                This action will also delete the following problems:
+              </Text>
+              <BulletList>
+                {toDelete.problems.map((problem, i) => {
+                  return (
+                    <li>
+                      <Text variant="subtitle1">
+                        {`Problem ${i}: ${problem.title}`}
+                      </Text>
+                    </li>
+                  );
+                })}
+              </BulletList>
+            </>
+          ) : (
+            <Text variant="subtitle1" gutterBottom>
+              This module has no problems associated with it
+            </Text>
+          )}
+
+          <Warning variant="body1">
+            This an action that can't be undone. Are you sure?
+          </Warning>
+        </DialogContent>
+
+        <Divider />
+
+        <Footer>
+          <FooterButton
+            onClick={(event) => {
+              event.stopPropagation();
+              handleClose();
+            }}
+            variant="contained"
+          >
+            Cancel
+          </FooterButton>
+
+          <FooterButton
+            onClick={(event) => {
+              event.stopPropagation();
+              handleDialogSubmit();
+            }}
+            variant="contained"
+            color="error"
+          >
+            Confirm
```

Comment:
I'd be interested to hear what others think, but I think it might be safer to make this button say "Delete" again just to make it absolutely clear what clicking that button would do

Reply:
Alright, I can do make that change next, its nothing big.